### PR TITLE
DZ60 and GMMK keymaps

### PR DESCRIPTION
**Pull request by dnedbaylo**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Madnace** [comment]
Hi ! Sorry i'm really new and nut at programing... I'm owning rn the GMMk full size and i wandted to know if it would be possible to add the complete firmware to qmk because i'm playing rythmic game and saw that the polling rate is 62.5hz wich means every keys  entries are sent to the pc every 16 milliseconds that is clearly impossible to play with this one...

Do you have some tips ? Can i still make my own firmware ?

File: keyboards/dz60/keymaps/dmitrynedbaylo/config.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#ifndef CONFIG_KEYMAP_H
+#define CONFIG_KEYMAP_H
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/config.h:2]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Also
```suggestion
#pragma once
```

File: keyboards/dz60/keymaps/dmitrynedbaylo/config.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/config.h:4]: This line is not actually needed, and should be removed. 

```suggestion
```

File: keyboards/dz60/keymaps/dmitrynedbaylo/config.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+#define TAPPING_TERM 200
+#define TAPPING_TERM_PER_KEY
+
+#define IGNORE_MOD_TAP_INTERRUPT
+#define PERMISSIVE_HOLD
+#define TAPPING_FORCE_HOLD_PER_KEY
+
+#endif
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/config.h:13]:
```suggestion
```

File: keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c [hunk 4]
```diff
@@ -0,0 +1,188 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    switch (keycode) {
+        case LCTL_T(KC_SPC):
+            return true;
+        default:
+            return false;
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t  state) {
+    switch (biton32(state)) {
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c:111]:
```suggestion
    switch (get_highest_layer(state)) {
```

File: keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            rgblight_setrgb (0x00,  0xFF, 0x00);
+            break;
+        case 5:
+            rgblight_setrgb (0x00,  0xFF, 0x00);
+            break;
+        case 7:
+            rgblight_setrgb (0x55,  0x55, 0x00);
+            break;
+        default: //  for any other layers, or the default layer
+            rgblight_setrgb (0x00,  0x00, 0xFF);
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c:122]:
```suggestion
            default_layer_state_set_user(default_layer_state);
```

File: keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+            break;
+    }
+    return state;
+}
+
+layer_state_t default_layer_state_set_user(layer_state_t state) {
+    switch (biton32(state)) {
```
Comment (drashna) [inline keyboards/dz60/keymaps/dmitrynedbaylo/keymap.c:129]:
```suggestion
    switch (get_highest_layer(state)) {
```

File: keyboards/gmmk/pro/keymaps/dmitrynedbaylo/config.h [hunk 8]
```diff
@@ -0,0 +1,13 @@
+#ifndef CONFIG_KEYMAP_H
+#define CONFIG_KEYMAP_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/gmmk/pro/keymaps/dmitrynedbaylo/config.h:4]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once
```

File: keyboards/gmmk/pro/keymaps/dmitrynedbaylo/config.h [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+#define TAPPING_TERM 200
+#define TAPPING_TERM_PER_KEY
+
+#define IGNORE_MOD_TAP_INTERRUPT
+#define PERMISSIVE_HOLD
+#define TAPPING_FORCE_HOLD_PER_KEY
+
+#endif
```
Comment (drashna) [inline keyboards/gmmk/pro/keymaps/dmitrynedbaylo/config.h:13]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
> Hi ! Sorry i'm really new and nut at programing... I'm owning rn the GMMk full size and i wandted to know if it would be possible to add the complete firmware to qmk because i'm playing rythmic game and saw that the polling rate is 62.5hz wich means every keys entries are sent to the pc every 16 milliseconds that is clearly impossible to play with this one...
> 
> Do you have some tips ? Can i still make my own firmware ?

This isn't exactly the correct place to ask this.  As a separate issue, or on the QMK Discord would be a better place for this. 

unfortunately, the full sized GMMK probably won't support QMK, period.

**Comment by dnedbaylo** [comment]
@drashna Sorry, I just realized I created a PR in qmk repo, not in my fork :facepalm: . Closing the PR.

**Pull request closed without merging**